File: examples/shapes/python/tests/test_circle.py
import pytest
import math
from .. import Circle

# Polytest Suite: circle

RADIUS = 7
circle = Circle(RADIUS)

# Polytest Group: circle

@pytest.mark.group_circle
def test_diameter():
    """A circle should be able to accurately calculate its diameter"""
    assert circle.diameter() == 14

@pytest.mark.group_circle
def test_radius():
    """A circle should be able to accurately calculate its radius"""
    assert circle.radius == 7

# Polytest Group: shape

@pytest.mark.group_shape
def test_perimeter():
    """A shape should be able to accurately calculate its perimeter (or circumference)"""
    assert circle.perimeter() == math.pi * RADIUS * 2

@pytest.mark.group_shape
def test_area():
    """A shape should be able to accurately calculate its area"""
    assert circle.area() == math.pi * RADIUS ** 2

File: examples/shapes/implementations/ruby/test/test_circle.rb
# frozen_string_literal: true

require "test_helper"

# circle
class TestCircle < Minitest::Test
  def setup
    @circle = Circle.new(7)
  end

  # Polytest Suite: circle

  # Polytest Group: circle

  def test_non_numeric
    assert_raises ArgumentError do
      Circle.new("Some radius")
    end
  end

  def test_diameter
    assert_equal 14, @circle.diameter
  end

  def test_radius
    assert_equal 7, @circle.radius
  end

  # Polytest Group: shape

  def test_perimeter
    assert_equal Math::PI * @circle.radius * 2, @circle.perimeter
  end

  def test_area
    assert_equal Math::PI * @circle.radius**2, @circle.area
  end
end
